src/notation/internal/engravingfontscontroller.cpp

```diff
@@ -0,0 +1,193 @@
+#include "engraving/internal/engravingfont.h"
+#include "log.h"
+
+#include <QDirIterator>
+#include <QFontDatabase>
+#include <QStandardPaths>
+
+using namespace mu::notation;
+
+void EngravingFontsController::init()
+{
+    auto musicFontsPath = configuration()->userMusicFontsPathChanged();
+    musicFontsPath.onReceive(this, [this](const muse::io::path_t&) {
+        scanAllDirectories();
+    });
+
+    scanAllDirectories();
+}
+
+void EngravingFontsController::scanAllDirectories() const
+{
+    engravingFonts()->clearExternalFonts();
+
+    // Standard locations as described in https://w3c.github.io/smufl/latest/specification/font-metadata-locations.html
+
+    // These standard location roughly match up with what the following returns, but some adjustments are needed.
+    QStringList standardLocations = QStandardPaths::standardLocations(QStandardPaths::GenericDataLocation).first(2);
+
+#if defined(Q_OS_WIN)
+    // On Windows, the second standard location returned by Qt is %ProgramData%, but we want %CommonProgramFiles%
+    QStringList globalFontsPaths {
+        standardLocations.first(),
+        qgetenv("CommonProgramFiles").replace("\\", "/")
+    };
+#elif defined(Q_OS_MACOS)
+    // MacOS is correctly handled by Qt
+    QStringList globalFontsPaths = standardLocations;
+#elif defined(Q_OS_LINUX)
+    // On Unix systems, we want $XDG_DATA_HOME (user-specific) and $XDG_DATA_DIRS (system-wide)
+    QStringList globalFontsPaths { qgetenv("XDG_DATA_HOME") };
+    globalFontsPaths.append(QString::fromLocal8Bit(qgetenv("XDG_DATA_DIRS")).split(':'));
+#endif
+
+    // The first location is the user-wide location, so we should iterate in reverse order so that
+    // user-specific fonts can override system-wide fonts
+    for (auto it = globalFontsPaths.rbegin(); it != globalFontsPaths.rend(); ++it) {
+        scanDirectory(*it + "/SMuFL/Fonts", false);
+    }
+
+    // Scan MuseScore-specific "private" location last so that it can override global fonts
+    scanDirectory(configuration()->userMusicFontsPath(), true);
+
+    QStringList musicFonts;
+
+    for (const auto& font : engravingFonts()->fonts()) {
+        musicFonts << QString::fromStdString(font->name());
+        musicFonts << QString::fromStdString(font->name() + " Text");
+    }
+
+    uiConfiguration()->setNonTextFonts(musicFonts);
+}
+
+void EngravingFontsController::scanDirectory(const muse::io::path_t& path, bool isPrivate) const
+{
+    using namespace muse::draw;
+
+    std::shared_ptr<IFontsDatabase> fdb = fontsDatabase();
+
+    QDirIterator iterator(path.toQString(), QDir::Dirs | QDir::NoDotAndDotDot | QDir::Readable);
+
+    while (iterator.hasNext()) {
+        iterator.next();
+        QString fontDir = iterator.filePath();
+        muse::io::path_t metadataPath;
+        QDirIterator fontFiles(fontDir, { "*.json" }, QDir::Files);
+
+        while (fontFiles.hasNext()) {
+            fontFiles.next();
+            metadataPath = fontFiles.filePath();
```

Comment:
So the last .json file wins?, No order of precendencd, of Fontname.json, fontname_metadate.json, metadata.json?

Reply:
The spec clearly states that the only standard filename is `fontname.json`. Is it really necessary to support non-standard installations? Also, the standard is very clear that the precedence is user-then-system.

(I realize that many on this thread already know this. I'm just explicitly bringing it out.)

Reply:
I would argue that there is no reason to not support other, non-standard-compliant metadata file names, especially since this is a reality already (see e.g. Bravura, Leland, many others...)

In any case, there should always only be one .json file in that directory, so I went with this approach.

The precedence should be handled correctly - see scanAllDirectories.

Reply:
There might be one for the 'normal' musical symbols fonr and another for the text font(s), like for Finale Ash, Finale Broadway (even 2, Text and Legacy Text), Finale Jazz (even 2, Text and Text Lowercase), Finale Maestro (even 5, Text Bold Italic, Text Bold, Text Italic, Text)

Reply:
Yes, but those would be in separate folders. The Finale fonts get installed like this (each folder has exactly one metadata json file):
![image](https://github.com/user-attachments/assets/63013d38-28ef-48fe-8b4c-c8b5f767bdaf)